tchannel/testing/vcr/cassette.py

```diff
@@ -39,20 +39,55 @@
 VERSION = 1
 
 
+def primitive_struct(obj):
+    data = {}
+    for field in obj.type_spec.fields:
+        attr = getattr(obj, field.name)
+        if attr is not None:
+            data[field.name] = primitive(attr, field.ttype_code)
+
+    return data
+
+
+def primitive_list(obj):
+    # I think this is because our objects aren't hydrating with the correct
+    # subtypes.
+    if obj and isinstance(obj[0], (str, dict)):
+        return obj
```

Comment:
This was because we're hydrating the request from yaml data (like `Request(**data)`) and so nested fields were still just dicts (instead of struct types).